patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -509,6 +509,16 @@
 +        }
 +    }
 +
++     /**
++     * Checked from {@link net.minecraft.client.multiplayer.PlayerControllerMP#onPlayerDestroyBlock(BlockPos pos) PlayerControllerMP.onPlayerDestroyBlock()}
++     * when a creative player left-clicks a block with this item.
++     * @return true if the given player can destroy specified block in creative mode with this item
++     */
++    public boolean canDestroyBlockInCreative(ItemStack stack, IBlockState state, EntityPlayer player)
```

Comment:
Yes, `World` and `BlockPos` are necessary here for use by blocks with tile entities.